fairlearn/reductions/moments/bounded_group_loss.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import pandas as pd
+import numpy as np
+from .moment import Moment
+from .moment import _REDUCTION_TYPE_LOSS_MINIMIZATION
+
+
+class LossMoment(Moment):
+    """A moment that quantifies a loss by group"""
+
+    def __init__(self, loss, no_groups=False):
+        super().__init__()
+        self.reduction_type = _REDUCTION_TYPE_LOSS_MINIMIZATION
+        self.reduction_loss = loss
+        self.no_groups = no_groups
+
+    def default_objective(self):
+        return AverageLossMoment(self.reduction_loss)
+
+    def init(self, dataX, dataA, dataY):
+        if self.no_groups:
+            dataA = dataY.apply(lambda y: "all")
+        super().init(dataX, dataA, dataY)
+        self.prob_attr = self.tags.groupby("protected_attribute").size() / self.n
```

Comment:
would be nice if "protected_attribute" is defined as a constant :-) we can do that later, too, if you don't want to do that now